rtbkit/common/exchange_connector.h

```diff
@@ -128,6 +128,11 @@ struct ExchangeConnector: public ServiceBase {
     */
     virtual std::string getBidSourceConfiguration() const;
 
+    /** Returns true if bid price is lower then the bidfloor price 
+    */
+
+    virtual bool bidFloorPrice(const Bid  & bid, const  std::vector<AdSpot> & imp, int & spotIndex) const;
```

Comment:
I would advocate for a more generic name. `bidFloorPrice` suggests that the function will only check the `bidfloor`. However, exchanges might have other requirements. meaning that determining if a bid is valid or not might depend on something else than the `bidfloor`. Since we already have functions like `getCampaignValidity` and `getCreativeValidity`, I think we could have exactly the same interface for a potential `getBidValidity` function. The default implementation would then check for the bidfloor. and the user will be free to override this function to inject his own logic (for example soft price for Smaato).